### PR TITLE
[gitter] Added gitter dashboard

### PR DESCRIPTION
**Pull request by imnitishng**
Added some basic visualizations for gitter backend as discussed [here](https://github.com/chaoss/grimoirelab-elk/pull/831).

Signed-off-by: Nitish Gupta <imnitish.ng@gmail.com>

### CODE REVIEW

**Comment by imnitishng** [comment]
Hi @valeriocos, here are the dashboards, I will write documentation and update these dashboards soon as our discussion progresses further.

There is a problem I am facing, the dashboards get uploaded successfully to Kibiter, however I get this error, the index pattern `gitter` I have saved in Kibiter is `gitter_e*`.
Can I get some help regarding this?

**Comment by valeriocos** [comment]
Thank you for the PR @imnitishng ! WRT the error please:
- replace all occurrences of `gitter-enriched` with `gitter` (in both files)
- replace `"title": "gitter_e*"` with `"title": "gitter"` (in the index pattern file)

This should fix the error. Let me know how it goes, thanks!

**Comment by imnitishng** [comment]
Hi @valeriocos, thank you for the quick response. I followed what you said but couldn't get the dashboard. Sorry I forgot to upload the image in the first comment.
I get this error 
Screenshot from 2020-04-11 22-09-43

The indexes I have used for gitter are
`gitter_raw` and `gitter_enriched_raw`, and I already have 2 indexes patterns manually made in Kibiter `gitter*` and `gitter_e*`.

**Comment by imnitishng** [comment]
@valeriocos, sure I will make the changes and add documentation here and update the How-to section. Thank you.

**Comment by imnitishng** [comment]
Hi @valeriocos, I have edited the dashboards as you said, I wanted to do this little change but I don't seem to get it.

How do I change URLs mentioned in table from plain text to this `Info+` link as used in Github pull request backlog?
Screenshot from 2020-04-12 12-01-14
Screenshot from 2020-04-12 12-00-37

**Comment by valeriocos** [comment]
Nice question! You need to update the format of the target attribute, see gif below.
Peek 12-04-2020 09-06

Please consider to add your question together with the answer to the [How to](https://github.com/chaoss/grimoirelab-sirmordred/blob/master/Getting-Started.md#how-to-) section of mordred.

**Comment by imnitishng** [comment]
Okay, thanks I will add this in the How-to section too.

**Comment by valeriocos** [comment]
Hi @imnitishng let me know when I can review this PR, thanks

**Comment by imnitishng** [comment]
@valeriocos, it is ready, I updated it. Please have a look.
I will update documentation post review.

**Comment by imnitishng** [comment]
Hi @valeriocos, sorry for the delay and mistakes. I have fixed them now. Please have a look.

**Comment by imnitishng** [comment]
Thanks for the review. I will update the PR ASAP.

**Comment by imnitishng** [comment]
@valeriocos, updated as you said.

File: docs/_panels/gitter.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+The Gitter panel shows information about messsages sent in a gitter room,
+over the course of it's age, gitter stores the text message sent alongwith
+the links in the form of username mentions, github or gitlab issues and
+pull requests which are automatically parsed as `@usename` and `#84` respectively
+by gitter. 
+
+Each user in a room is uniquely identified by a username and a user-id. This
+information is used to identify the most active and mentioned users.
+Similarly the links shared are identified as github issues or pull requests
+or categorized into URL hostnames if they belong to external sources.
```
Comment (valeriocos) [inline docs/_panels/gitter.md:20]: external or other sources?

File: docs/_panels/gitter.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Weekly, hourly and daily activity of the room is put into perspective by
+widgets displaying the number of messages sent or unique users active in
+the for selected time frame.
```
Comment (valeriocos) [inline docs/_panels/gitter.md:24]: in the for selected time frame -> in the selected time range (to be consistent with the description below)

File: docs/_panels/gitter.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+title: Gitter
+description: metrics focused on Gitter messages.
+author: Nitish Gupta <imnitish.ng@gmail.com>
+screenshot: sigils/gitter.png
+created_at: 2020-04-18 (YYYY-MM-DD)
+grimoirelab_version: 0.2.0
+layout: panel
+---
+
+The Gitter panel shows information about messsages sent in a gitter room,
```
[the remark this answers is not in the document]
Comment (imnitishng) [reply docs/_panels/gitter.md:11]: Hi @valeriocos 
Yes, GitLab issues work fine.
But in order to support merge requests, a small logic change should be made in the Gitter Enricher.
How do you want me to go about this?

Also, if we support merge requests, we might need to add another panel for them. Which will usually be empty incase the room belongs to an organization which does not have it's projects on GitLab.

And considering the current ELK implementation, the gitlab issues are also shown in GitHub issues visualization (mixed). What about that?
Do we add 2 more visualizations for GitLab issues and mr along with ELK logic change, or leave it as they are.

File: docs/_panels/gitter.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+over the course of it's age, gitter stores the text message sent alongwith
+the links in the form of username mentions, github or gitlab issues and
+pull requests which are automatically parsed as `@usename` and `#84` respectively
+by gitter. 
+
+Each user in a room is uniquely identified by a username and a user-id. This
+information is used to identify the most active and mentioned users.
+Similarly the links shared are identified as github issues or pull requests
```
[the remark this answers is not in the document]
Comment (imnitishng) [reply docs/_panels/gitter.md:19]: Sure. Will fix.

File: docs/_panels/gitter.md (hunk 1 above)
Comment (imnitishng) [reply docs/_panels/gitter.md:20]: Other sources, other websites apart from gitlab or github. (stackoverflow, twitter etc)

File: docs/_panels/gitter.md (hunk 2 above)
Comment (imnitishng) [reply docs/_panels/gitter.md:24]: Okay thanks, will fix.

File: docs/_panels/gitter.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (valeriocos) [reply docs/_panels/gitter.md]: Let's finish this first iteration and then we can focus on the improvements.
Please open an issue in ELK to document the change you suggest.

Thanks!

File: docs/_panels/gitter.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (valeriocos) [reply docs/_panels/gitter.md]: Thanks!

File: docs/_panels/gitter.md (hunk 1 above)
Comment (valeriocos) [reply docs/_panels/gitter.md]: ok, thanks for the clarification. I would write it in this way:
```
.. if they belong to other sources (e.g., StackOverflow, Twitter).
```

File: docs/_panels/gitter.md (hunk 2 above)
Comment (valeriocos) [reply docs/_panels/gitter.md]: Thanks!

File: docs/_panels/gitter.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (imnitishng) [reply docs/_panels/gitter.md]: Sure I will.

**Comment by imnitishng** [comment]
@valeriocos, I have implemented the changes. Please have a look. We should be good to go now.

**Comment by valeriocos** [comment]
I still the scroll in the markdown:
captura_361

Can you get an output like the one below?
captura_362

**Comment by imnitishng** [comment]
Sorry for the mistake @valeriocos.
But I think this does depend on user's monitor size. Because when I created the edits, I made sure there were no scroll bars visible ([screenshot added in documentation](https://raw.githubusercontent.com/imnitishng/grimoirelab-sigils/gitter_dash/docs/assets/images/screenshots/sigils/gitter.png)), IDK how they come up on your screen.
Anyway, I am fixing it. Thanks for pointing out.

**Comment by valeriocos** [comment]
No worries @imnitishng ! It is still not fixed, can you try with the following files below?
[gitter.zip](https://github.com/chaoss/grimoirelab-sigils/files/4510176/gitter.zip)

Please check also the commit message, since everything appears on the same line

**Comment by imnitishng** [comment]
Fixed it. I hope it's all good now.

**Pull request closed without merging**